### PR TITLE
Replaced eval script with native command to read from redis. Native c…

### PR DESCRIPTION
**Pull request by alexandraprisecarutrimble**
Replaced eval script with native command to read from redis. Native commands have better performance compared to EVAL/EVALSHA

Summary of the changes:
 - In order to read from redis cache, a script was evaluated which got data for a key along with expiration information
 - This can be done using methods provided by the StackExchange.Redis nuget package which uses native commands
-  EVAL/EVALSHA have a greater impact on performance compared to native commands and it lead to performance spikes on real use case scenarios

Sample of redis monitoring before applying the changes:
evalscriptsforreading
Sample of redis monitoring after applying the changes:
nativecommandsforreading

Addresses #3632

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/extensions?pullRequest=3643) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/extensions?pullRequest=3643) before we can accept your contribution.<br/><br/>:x: alexandraprisecarutrimble [sign now](https://cla.dotnetfoundation.org/dotnet/extensions?pullRequest=3643)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/extensions?pullRequest=3643) it.</sub>

File: src/Caching/StackExchangeRedis/src/RedisExtensions.cs [hunk 1]
```diff
@@ -8,31 +8,19 @@ namespace Microsoft.Extensions.Caching.StackExchangeRedis
...
             params string[] members)
         {
-            var result = await cache.ScriptEvaluateAsync(
-                HmGetScript,
-                new RedisKey[] { key },
-                GetRedisMembers(members));
-
             // TODO: Error checking?
-            return (RedisValue[])result;
+            return await cache.HashGetAsync(key, GetRedisMembers(members)).ConfigureAwait(false);
```
Comment (BrennanConroy) [inline src/Caching/StackExchangeRedis/src/RedisExtensions.cs:23]: Nice, a missing `ConfigureAwait(false)` :+1:

**Review by BrennanConroy — APPROVED** [review]

**Comment by Tratcher** [comment]
Note this has only been approved for merging into the master (6.0) branch so far. We're going to need more information to justify patching this in 3.1. Just how bad of an issue is this for you?

**Comment by alexandraprisecarutrimble** [comment]
Hello, @Tratcher  
At the moment, we believe the current implementation of the redis cache reading operation is the root cause of the performance spikes in our services. Generally we have very good performance, ~2-5ms response times, but there are performance spikes which can take up to a few seconds. 
We are about to release one of our products next month and we cannot afford having these performance spikes, nor can we wait until version 6.0 to have this implementation available. After this update, we didn't see any major spikes, no matter how much we stressed the services, we even saw a performance boost, the average time of reading from an AWS Elasticache Redis cluster being now <1ms.
I believe this change can be beneficial for everyone and it won't disrupt the current workflows. 
If this change won't be integrated in 3.1, we will most likely stop using this nuget and use a temporary nuget that includes this change. We would like to avoid that and use this official nuget as the single source of truth.
Please let me know what you think and whether or not this change can be patched to 3.1 or not, so we know what steps to make.
Thanks!

**Comment by Tratcher** [comment]
Note even if this is approved for 3.1, those patches only happen once per month which may not meet your schedule.

**Comment by Tratcher** [comment]
@BrennanConroy can you fill out the shiproom template and get this into the next 3.x milestone?

**Comment by BrennanConroy** [comment]
#### Description

The redis cache implementation uses a lua script when getting keys, this is slower than using a native redis function and because we aren't doing anything custom with the lua script we should just be using the native function. Customer noticed spikes that increased latency in their app and found that it was related to the redis cache and after investigation found that switching to the native function fixed the issue.

#### Customer Impact

Issue reported by customer in https://github.com/dotnet/extensions/issues/3632. Causes spikes that increase latency.

Potential workaround would be to implement another caching implementation on top of redis caching that does in-memory lookups and falls back to redis after X time or every X calls. This could lead to invalid cache values though and is almost the same as just forking the code and writing your own caching.

#### Regression?

No

#### Risk

Low

**Comment by Pilchie** [comment]
This is approved, but need to wait to merge until the branch is open for the next release.  @BrennanConroy - we'll also need to make sure we get this into 5.0.1 servicing as well.

**Pull request merged**